src/frontend/config.ts

```diff
@@ -1,24 +1,134 @@
     ConfigElements.services,
     ConfigElements.contracts,
   ]);
+
   return config.contracts.map((contract) => {
     return {
-      name: `${contract.name} - ${
+      id: contract.id,
+      name: contract.name,
+      vendorName:
         config.services.find((svc) => svc.vendor_id === contract.vendor_id)
-          ?.vendor_name
-      }`,
-      contract_id: contract.id,
+          ?.vendor_name ?? "",
+    };
+  });
+};
+
+export const getContractAndVendorName = async (
+  contractId: string
+): Promise<{ vendorName: string; contractName: string }> => {
+  const config = await makeCtxConfig([
+    ConfigElements.services,
+    ConfigElements.contracts,
+  ]);
+
+  const contract = config.contracts.find(
+    (contract) => contract.id === contractId
+  );
+
+  if (contract === undefined) {
+    throw new Error("No contract found");
+  }
+
+  const vendorName =
+    config.services.find((svc) => svc.vendor_id === contract.vendor_id)
+      ?.vendor_name ?? "";
+
+  return { vendorName, contractName: contract.name };
+};
+
+export const getContractPrettyName = async (id: string): Promise<string> => {
+  const config = await makeCtxConfig([
+    ConfigElements.services,
+    ConfigElements.contracts,
+  ]);
+  const contract = config.contracts.find((contract) => contract.id === id);
+  if (contract === undefined) {
+    throw new Error("No contract found");
+  }
+  return `${contract.name} - ${
+    config.services.find((svc) => svc.vendor_id === contract.vendor_id)
+      ?.vendor_name
+  }`;
+};
+
+export const getContractName = async (id: string): Promise<string> => {
+  const config = await makeCtxConfig([
+    ConfigElements.services,
+    ConfigElements.contracts,
+  ]);
+  const contract = config.contracts.find((contract) => contract.id === id);
+  if (contract === undefined) {
+    throw new Error("No contract found");
+  }
+  return contract.name;
+};
+
+const isCompleteDatum = (datum: Datum): datum is { VarCharValue: string } =>
+  !!datum.VarCharValue;
+
+const isCompleteDataArray = (
+  data: Datum[] | undefined
+): data is Array<{ VarCharValue: string }> =>
+  !!data?.every((datum) => datum !== undefined && isCompleteDatum(datum));
+
+const athena = new AthenaClient({ region: "eu-west-2" });
+
+const QUERY_WAIT = 250; // 0.25 second
+
+const months = [
+  "Jan",
+  "Feb",
+  "Mar",
+  "Apr",
+  "May",
+  "Jun",
+  "Jul",
+  "Aug",
+  "Sep",
+  "Oct",
+  "Nov",
+  "Dec",
+];
+
+// Write getContractData which will return the display name and all the months/years
+export const getContractPeriods = async (
+  contractName: string
+): Promise<Array<{ month: string; year: string; prettyMonth: string }>> => {
+  if (process.env.QUERY_RESULTS_BUCKET === undefined)
+    throw new Error("No QUERY_RESULTS_BUCKET defined in this environment");
+
+  const fetchDataSql = `SELECT DISTINCT month, year FROM "${process.env.DATABASE_NAME}".btm_monthly_extract WHERE contract_name LIKE '${contractName}' ORDER BY year DESC, month DESC`;
```

Comment:
What is the benefit of using contract_id over contract_name?

Reply:
Spoken with Mark S on this and will use contract_id.

Reply:
We did talk about this when we worked on this part. It would be good to be able to query this table by contract ID because then we could run the fetch for the config and the fetch for the data in parallel. It's one the backlog https://govukverify.atlassian.net/browse/BTM-629

Reply:
We've done this now as part of the next push.